packages/chat-core/src/index.ts

```diff
@@ -4,113 +4,343 @@ import { equals, adjust, omit } from "ramda";
-
+/**
+ * A message from the user
+ *
+ * See also:
+ * - {@link BotResponse}
+ * - {@link FailureMessage}
+ * - {@link Response}
+ *
+ */
 export interface UserResponse {
+  /**
+   * The type of the response is `"bot"` for bot and `"user"` for user, and "failure" for failure.
+   */
   type: "user";
+  /**
+   * When the response was received
+   */
   receivedAt: Time;
+  /**
+   * The payload of the response
+   */
   payload: UserResponsePayload;
 }
 
+/**
+ * The payload of the user response
+ */
 export type UserResponsePayload =
   | {
+      /**
+       * Set when `sendText` is called.
+       */
       type: "text";
+      /**
+       * The user's message
+       */
       text: string;
+      /**
+       * [Context](https://docs.studio.nlx.ai/workspacesettings/documentation-settings/settings-context-attributes) for usage later in the intent.
+       */
       context?: Context;
     }
   | {
+      /**
+       * Set when `sendChoice` is called.
+       */
       type: "choice";
+      /**
+       * The `choiceId` passed to `sendChoice`
+       * Correlates to a `choiceId` in the {@link BotResponse}'s `.payload.messages[].choices[].choiceId` fields
+       */
       choiceId: string;
+      /**
+       * [Context](https://docs.studio.nlx.ai/workspacesettings/documentation-settings/settings-context-attributes) for usage later in the intent.
+       */
       context?: Context;
     }
   | ({
+      /**
+       * Set when `sendStructured` is called.
+       */
       type: "structured";
+      /**
+       * [Context](https://docs.studio.nlx.ai/workspacesettings/documentation-settings/settings-context-attributes) for usage later in the intent.
+       */
       context?: Context;
     } & StructuredRequest);
 
 // Failure message
 
+/**
+ * A failure message is received when the NLX api is unreachable, or sends an unparsable response.
+ */
 export interface FailureMessage {
+  /**
+   * The type of the response is `"bot"` for bot and `"user"` for user.
+   */
   type: "failure";
+  /**
+   * The payload only includes an error message.
+   */
   payload: {
+    /**
+     * The error message is either the default, or the `failureMessage` set in the {@link Config}.
+     */
     text: string;
   };
+  /**
+   * When the failure occurred.
+   */
   receivedAt: Time;
 }
 
+/**
+ * A response from the bot or the user.
+ */
 export type Response = BotResponse | UserResponse | FailureMessage;
 
+/**
+ * The time value in milliseconds since midnight, January 1, 1970 UTC.
+ */
 export type Time = number;
 
 // Config and state
 
+/**
+ * @hidden
+ */
 export type Environment = "production" | "development";
 
+/**
+ * The config to create a conversation.
+ * `botUrl` and
+ */
 export interface Config {
+  /**
+   * Fetch this from the bot's Deployment page.
+   */
   botUrl: string;
+  /**
+   * The conversation ID. If not set, a new conversation will be started.
+   */
   conversationId?: string;
+  /**
+   * TODO I don't know what this is
```

Comment:
AFAICT this is to help user's find their own user ids? E.g. this is not a NLX User id but rather a an arbitrary, optional user id in a customer's system.

Reply:
Basically, if you give us a user ID, the conversation logs will be searchable by it if the privacy agreement allows. It will also be usable inside the canvas (`{System.userId}` inside messages, payloads etc.).